test/Integration.t.sol

```diff
@@ -0,0 +1,311 @@
+
+}
+
+contract GeneralisedTests is CapAutomatorIntegrationTestsBase {
+
+    using ReserveConfiguration for DataTypes.ReserveConfigurationMap;
+    using WadRayMath           for uint256;
+
+    function test_E2E_increaseBorrow() public {
+        for(uint256 i; i < assets.length; i++) {
+            DataTypes.ReserveData memory reserveData = pool.getReserveData(assets[i]);
+
+            uint256 preIncreaseBorrowCap = reserveData.configuration.getBorrowCap();
+            if (preIncreaseBorrowCap == 0) {
+                continue;
+            }
+
+            uint256 currentBorrow = ERC20(reserveData.variableDebtTokenAddress).totalSupply() / 10 ** ERC20(reserveData.variableDebtTokenAddress).decimals();
+
+            uint256 preIncreaseBorrowGap = preIncreaseBorrowCap - currentBorrow;
+
+            uint256 newMaxCap = preIncreaseBorrowCap * 2;
+            uint256 newGap    = preIncreaseBorrowGap * 2;
+
+            vm.prank(SPARK_PROXY);
+            capAutomator.setBorrowCapConfig({
+                asset:            assets[i],
+                max:              newMaxCap,
+                gap:              newGap,
+                increaseCooldown: 12 hours
+            });
+
+            capAutomator.exec(assets[i]);
+
+            uint256 postIncreaseBorrowCap = pool.getReserveData(assets[i]).configuration.getBorrowCap();
+
+            assertEq(postIncreaseBorrowCap, currentBorrow + newGap);
+        }
+    }
+
+    function test_E2E_decreaseBorrow() public {
+        for(uint256 i; i < assets.length; i++) {
+            DataTypes.ReserveData memory reserveData = pool.getReserveData(assets[i]);
+
+            uint256 preDecreaseBorrowCap = reserveData.configuration.getBorrowCap();
+            if (preDecreaseBorrowCap == 0) {
+                continue;
+            }
+
+            uint256 currentBorrow = ERC20(reserveData.variableDebtTokenAddress).totalSupply() / 10 ** ERC20(reserveData.variableDebtTokenAddress).decimals();
+
+            uint256 preDecreaseBorrowGap = preDecreaseBorrowCap - currentBorrow;
+
+            uint256 newGap = preDecreaseBorrowGap / 2;
+
+            vm.prank(SPARK_PROXY);
+            capAutomator.setBorrowCapConfig({
+                asset:            assets[i],
+                max:              preDecreaseBorrowCap,
+                gap:              newGap,
+                increaseCooldown: 12 hours
+            });
+
+            capAutomator.exec(assets[i]);
+
+            uint256 postDecreaseBorrowCap = pool.getReserveData(assets[i]).configuration.getBorrowCap();
+
+            assertEq(postDecreaseBorrowCap, currentBorrow + newGap);
+        }
+    }
+
+    function test_E2E_increaseSupply() public {
+        for(uint256 i; i < assets.length; i++) {
+            DataTypes.ReserveData memory reserveData = pool.getReserveData(assets[i]);
+
+            uint256 preIncreaseSupplyCap = reserveData.configuration.getSupplyCap();
+            if (preIncreaseSupplyCap == 0) {
+                continue;
```

Comment:
If we're gonna make generalized tests like these we should perform assertions on both options instead of skipping